schema/overview.md

```diff
@@ -0,0 +1,68 @@
+IPLD Schemas Overview
+=====================
+
+
+Goals
+-----
+
+1. Provide a reasonable, easy to use, general type system for declaring
+  useful properties about data.
+2. Compose nicely over IPLD Data Model.  Schemas should be usable with any
+  data format you can build an IPLD Codec for.
+3. Be efficient to verify (e.g. predictable; roughly linear in the size of
+  the data and schema; and absolutely not turing complete).
+4. Be language-agnostic: many compatible implementations of the schema
+  checker tools should exist, as well as bindings for every language.
+5. Assist rather than obstruct migration.  (We expect data to exist from
+  *before* the current schemas; we need to work well on this case.)
+6. Be a home for collaboration and documentation!  Schemas should provide a
+  natural place for API documentation, and be a reasonable design literature.
+
+In total: **Make developing decentralized & distributed applications
+easier, clearer, and more robust!**
+
+
+Features
+--------
+
+Well-understood basics (sum types, product types, some specific recursive types):
+
+- typed maps (e.g. `{Foo:Bar}`)
+- typed arrays (e.g. `[Foo]`)
+- typedef'd primitives (e.g. `type Foo int`)
+- structs (e.g. `struct{ f Foo; b Bar }`)
+- discriminated unions (several styles)
+- enumerations (e.g. `["Foo" | "Bar" | "Baz"]`)
+
+Bonus features:
+
+- simple syntax for 'maybe' (e.g. `struct{ f nullable Foo }` or `[nullable Foo]`)
+- clear syntax for non-required fields (e.g. `struct { f optional Foo }`)
+
+All of this maps over the IPLD Data Model:
+
+- every type has a representation;
+- e.g. maps are represented as maps; structs are also represented as maps!
+- representations can specify aliases for how to serialize things; e.g. provide
+  shorter serial strings to use in place of longer struct names.
+- representations are customizable, including mapping a type onto a different
+  representation kind entirely!  e.g. a struct can be declared to have a
```

Comment:
you're using "kind" here so I think that suggests you should use it with or instead of "primitive" above (although I quite like using it _with_ "primitive" as it helps explain "kind" which is bound to be unfamiliar to some audience)